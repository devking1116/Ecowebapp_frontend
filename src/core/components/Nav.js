/* @flow */
import React from 'react';
import reactCSS from 'reactcss';

import { HeaderMenu } from '../../core';

type Props = {
  children?: any,
};

/**
 * The header for the website.
 */
export default function Nav({ children, ...otherProps }: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      nav: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        width: 700,
        height: '100%',
      },
    },
  });

  return (
    <nav style={styles.nav} {...otherProps}>
      <HeaderMenu>
        {children}
      </HeaderMenu>
    </nav>
  );
}
