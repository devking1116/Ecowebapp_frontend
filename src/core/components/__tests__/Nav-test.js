import React from 'react';
import renderer from 'react-test-renderer';

import Nav from '../Nav';

test('it displays its children in a <nav /> tag.', () => {
  const component = renderer.create(
    <Nav>
      <div>Test Nav Item 0</div>
      <div>Test Nav Item 1</div>
      <div>Test Nav Item 2</div>
    </Nav>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
