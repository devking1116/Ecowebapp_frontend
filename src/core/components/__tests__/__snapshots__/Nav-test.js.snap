exports[`test it displays its children in a <nav /> tag. 1`] = `
<nav
  style={
    Object {
      "WebkitJustifyContent": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": 700,
    }
  }>
  <div
    className="ui inverted secondary menu"
    style={
      Object {
        "height": "100%",
        "marginBottom": 0,
        "marginTop": 0,
      }
    }>
    <div
      className="item"
      onClick={[Function]}>
      <div>
        Test Nav Item 0
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <div>
        Test Nav Item 1
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <div>
        Test Nav Item 2
      </div>
    </div>
  </div>
</nav>
`;
